datm_in

```diff
@@ -1,5 +1,5 @@
 &datm_nml
```

Comment:
Should `flds_presaero` and `flds_presndep` be set to `.false.` now that we've removed those streams?

Reply:
In the current configuration, the flags `flds_preasero` and `flds_presndep` are set to `.true.` in `datm_in`. These flags are meant to control the inclusion of aerosol and lnd streams in the field list. However, even after removing the corresponding aerosol streams from the `datm.streams.xml` and `config.yaml` files, the code does not return any errors as expected.

It is evident that when `flds_presaero = .true.`, the aerosol streams are added to the field list in the source code [here](https://github.com/ESCOMP/CDEPS/blob/f027aa64285fb9ddad9be5c5837a6e6e279e6051/datm/datm_datamode_jra_mod.F90#L113-L118) and advertised [here](https://github.com/ESCOMP/CDEPS/blob/f027aa64285fb9ddad9be5c5837a6e6e279e6051/datm/datm_datamode_jra_mod.F90#L132)

However, even after removing the aerosol streams from the data streams configuration files, the mediator log shows that aerosol fields are still being advertised, which is unexpected behavior. 

``` 
(Advertise Fields):Fr_atm: Faxa_bcph 
(Advertise Fields):Fr_atm: Faxa_ocph
(Advertise Fields):Fr_atm: Faxa_dstwet
(Advertise Fields):Fr_atm: Faxa_dstdry 
```

I am checking the reason for this behaviour, since the removal of aerosol streams in the datm xml file with the flag `flds_presaero = .true` should return some error.

Reply:
Great - thanks for looking into this - I think it's important that we understand why we don't get the behaviour we expect

Reply:
Thanks @dougiesquire. Upon investigation into the code and the mediator log, it has been observed that the aerosol streams, specifically `Faxa_bcph, Faxa_ocph, Faxa_dstwet, and Faxa_dstdry`, are indeed added to the field list and advertised when the `flds_presaero` flag is set to `.true.`. Similarly, the land streams are also part of the field list and advertised when `flds_presndep` is set to `.true.`.

However, despite being included in the field list and advertised, there seems to be no utilization or call for these aerosol and land streams by other components. The mediator log indicates that these stream variables are not mapped into other components, which explains why there are no errors when we set `flds_presaero = .false.` and `flds_presndep = .false.` in the `datm_in` file.

```
 mapping atm->ocn Sa_u via patch_uv3d with one normalization
 mapping atm->ocn Sa_v via patch_uv3d with one normalization
 mapping atm->ocn Sa_z via bilnr with one normalization
 mapping atm->ocn Sa_tbot via bilnr with one normalization
 mapping atm->ocn Sa_pbot via bilnr with one normalization
 mapping atm->ocn Sa_shum via bilnr with one normalization
 mapping atm->ocn Sa_ptem via bilnr with one normalization
 mapping atm->ocn Sa_dens via bilnr with one normalization
 mapping atm->ocn Faxa_swnet via consf with one normalization
 mapping atm->ocn Faxa_rainc via consf with one normalization
 mapping atm->ocn Faxa_rainl via consf with one normalization
 mapping atm->ocn Faxa_snowc via consf with one normalization
 mapping atm->ocn Faxa_snowl via consf with one normalization
 mapping atm->ocn Faxa_lwdn via consf with one normalization
 mapping atm->ocn Faxa_swndr via consf with one normalization
 mapping atm->ocn Faxa_swvdr via consf with one normalization
 mapping atm->ocn Faxa_swndf via consf with one normalization
 mapping atm->ocn Faxa_swvdf via consf with one normalization
 mapping atm->ocn Sa_pslv via bilnr with one normalization
```


To clarify the configuration and avoid any confusion, I am setting `flds_presaero = .false.` and `flds_presndep = .false.` in the `datm_in` file to ensure that these aerosol and land streams are not included in the field list and are not advertised.

Reply:
Interesting. Is this the case even when streams are provided for `presaero` and `presndep`?

Reply:
Yes, that's correct

Reply:
Just checking I understand what you're saying: it's currently _not possible_ to use `presaero` and `presndep` forcing? Worse, it's possible to set things up so that it looks like `presaero` and `presndep` forcing is being applied, but the data is never actually being used?

Reply:
Mediator log reveals aerosol streamfiles are being mapped to the ice component, regardless of the presence of input aerosol files. 

```
 mapping atm->ice Faxa_bcph via consf with one normalization
 mapping atm->ice Faxa_dstwet via consf with one normalization
 mapping atm->ice Faxa_dstdry via consf with one normalization
```
This is unexpected behaviour,  since the code should return some error if there is no input files for the aerosol streams. For clarity, `flds_presaero` and `flds_presndep` are set to `.false.` in the current configuration and also `tr_aero` in `ice_in` need to be set to `.false.`.